src/lib.rs

```diff
@@ -440,7 +440,7 @@ pub trait Strategy<'a> {
 /// A strategy that uses polling to efficiently wait for an event.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
 pub struct NonBlocking<'a> {
-    _marker: PhantomData<Context<'a>>,
+    _marker: PhantomData<&'a &'a mut ()>,
```

Comment:
Could you leave a comment as to why you are using `&'a &'a mut ()` instead of `&'a ()` here?